hooks/test/browser/combinations.test.js

```diff
@@ -300,7 +300,9 @@ describe('combinations', () => {
 		]);
 	});
 
-	it('should run effects child-first even for children separated by memoization', () => {
+	// TODO: I actually think this is an acceptable failure, because we update child first and then parent
+	// the effects are out of order
```

Comment:
Agreed - this is updating via a top-level render, which is super uncommon. Normally the child and parent would update via setState or forceUpdate(), which would mean they'd go into the update queue and that queue would flush them inner-to-outer. I wonder if we could keep this test and make it pass by switching to invoking `setState()` on both components rather than `rerender()` of the outer?

Reply:
Well this is impossible to bypass in tests at least, we render `Parent` but `Child` is memoized meaning we won't enter that. We exit diffing and push this one on the `afterPaintEffects` queue, then we do a new setState and push that one on the queue

Reply:
@developit @JoviDeCroock (playing catch up over the last month 😅) This test was added to regress this PR and the related issue: #3354

Are we saying that this test doesn't accurately represent the issue, that we longer think the issue is a bug, or something else?

Reply:
I would be inclined to label this as an edge case; it's hard to differentiate between these updates because it are not trickle down renders. Fixing this would imply us inverting the render queue completely. This behavior seems inline with React too